src/messaging/ExchangeContext.cpp

```diff
@@ -146,6 +207,15 @@ void ExchangeContext::DoClose(bool clearRetransTable)
     }
     mDelegate = nullptr;
 
+    // Flush any pending CRMP acks
+    mReliableMessageContext.FlushAcks();
+
+    // Clear the CRMP retransmission table
+    if (clearRetransTable)
+    {
+        mExchangeMgr->GetReliableMessageMgr()->ClearRetransTable(&mReliableMessageContext);
+    }
+
```

Comment:
Same as above.

Would you please extract these logics into a function in ReliableMessageContext or ReliableMessageManager class ?

The code is totally fine, and reasonable. But putting the code here will significantly distract one's mind when reading the code.

Preferable you can use add a `ReliableMessageContext::Close` function

Reply:
Some logic with this function such as CancelResponseTimer() is not CRMP, this is for another response mechanism operating on the protocol level. 

We can revisit those optimization after the first integration.